### ISSUE TITLE
RFC: Provide a web-based client for the FauxPilot server

### ISSUE THREAD

**Issue by leemgs** [moyix/fauxpilot#88]
## Background

As everyone knows, Faxupilot Server has the following advantages over the commercial Copilot service:
1. Since it is an **open source** project, the little cord is modifiable and used.
2. A thinking server can only be operated on its own server, **not in the cloud**.
3. It is **free** to use because it is accessible.

Using the FauxPilot's installation guide, we can operate our own inferencing server if we have an Nvidia GPU (The version of Compute Capability must be 6.0 at least.) And, as indicated in the handbook, there are other ways to construct a client.
Using the methods given in the manual, we can transmit statements to the FauxPilot server and obtain source code that has been automatically generated by FauxPilot server.

In general, during the period of development, we frequently try to test the statements created by the developers for pre-test and evaluation. I believe that these pre-tests will improve development efficiency and productavity when using a web-based client for the FauxPilot Server. :)

## Proposal

We can easily and conveniently test the inference results run by the FauxPilot server using the REST API and web programming languages. In this case, the user only needs a web browser. And our colleagues can immediately participate in the test and feed back with only the web browser. So I would like to provide a web-based FauxPilot client in order for users and developers of the FauxPilot project to read more convenient and immediately while performing various tests.

## User Interface of Web-based client (Draft)

The pictures below show the GUI examples of proposed web-based client.
The first picture is the input screen. And the second picture is the screen showing the output results.

TBA. :)

**Comment by moyix** [comment]
Is there a web client actually implemented right now, or are these UI mockups meant to show what it would look like?

Another option for building an client webapp would be something like [Gradio](https://gradio.app/)

**Comment by leemgs** [comment]
> Is there a web client actually implemented right now,

Yes, I will submit the PR for review immediately after conducting a self-evaluation.

**Comment by azeemba** [comment]
I think a web client is a great idea but I do have some concerns with the implementation in #91

This project is mostly python but the implementation in #91 adds php dependencies. I think @moyix suggestion of using [Gradio](https://gradio.app/) makes a lot of sense, especially since the resulting script will probably only take up one file. Since the implementation will be so small, it should reduce the concerns about the maintenance burden as well.

@moyix are you interested in adding a gradio-based client to this repo? Or would you still prefer for it to be external due to the maintenance concerns?

**Comment by leemgs** [comment]
Please refer to the latest comment on PR 91.

**Comment by leemgs** [comment]
This issue will be closed. :)
